gulpfile.babel.js

```diff
@@ -0,0 +1,203 @@
+  config.sass.options = config.environment === 'production' ? {outputStyle: 'compressed'} : null;
+
+  config.dir = {
+    src: {},
+    test: {},
+    out: {},
+  };
+  config.dir.src.css = `${config.sourceRoot}/css`;
+  config.dir.src.sass = `${config.dir.src.css}/sass`;
+  config.dir.src.images = `${config.sourceRoot}/images`;
+  config.dir.src.fonts = `${config.sourceRoot}/fonts`;
+  config.dir.src.templates = `${config.sourceRoot}/_patterns`;
+  config.dir.src.js = `${config.sourceRoot}/js`;
+
+  config.dir.test.sass = `${config.testRoot}/sass`;
+
+  config.dir.out.css = `${config.exportRoot}/css`;
+  config.dir.out.sass = `${config.dir.out.css}/sass`;
+  config.dir.out.sassVendor = `${config.dir.out.css}/sass/vendor`;
+  config.dir.out.images = `${config.exportRoot}/images`;
+  config.dir.out.fonts = `${config.exportRoot}/fonts`;
+  config.dir.out.templates = `${config.exportRoot}/templates`;
+
+  config.files = {
+    src: {},
+    test: {},
+    out: {},
+  };
+  config.files.src.css = [
+    `${config.dir.src.css}/**/*.css`,
+    `${config.dir.src.css}/**/*.map`,
+    `!${config.dir.src.css}/pattern-scaffolding.css`,
+    `!${config.dir.src.css}/sass/**/*`,
+  ];
+  config.files.src.sass = [
+    `${config.dir.src.sass}/**/*.scss`,
+    `!${config.dir.src.sass}/vendor/**/*`,
+  ];
+  config.files.src.sassEntryPoint = `${config.dir.src.sass}/${invocationOptions.sassEntryPoint}`;
+  config.files.src.sassVendor = [
+    `${config.dir.src.sass}/vendor/**/*.scss`,
+    `${config.dir.src.sass}/vendor/**/*.css`,
+    `${config.dir.src.sass}/vendor/**/LICENSE.*`,
+  ];
+  config.files.src.images = [`${config.dir.src.images}/*`, `${config.dir.src.images}/**/*`];
+  config.files.src.fonts = [`${config.dir.src.fonts}/*`, `${config.dir.src.fonts}/**/*`];
+  config.files.src.templates = [`${config.dir.src.templates}/*.twig`, `${config.dir.src.templates}/**/*.twig`];
+  config.files.src.derivedConfigs = [
+    `${config.dir.src.sass}/variables/**/*`,
+    `${config.dir.src.js}/derivedConfig.json`,
+  ];
+
+  config.files.test.sass = `${config.dir.test.sass}/**/*.spec.scss`;
+  config.files.test.sassTestsEntryPoint = `${config.dir.test.sass}/test_sass.js`;
+
+  config.files.out.cssFilename = invocationOptions.cssOutFilename;
+
+  return config;
+
+};
+
+const config = buildConfig(process.argv, 'source', 'test', 'export');
+
+// Builders
+
+const cleanSharedConfig = () => del(config.files.src.derivedConfigs);
+
+export const distributeSharedConfig = gulp.series(cleanSharedConfig, distributeConfig);
+
+export const lintSass = () => {
+  if (!config.sass.linting) {
+    console.info('Skipping sass:lint');
+    return Promise.resolve();
+  }
+
+  const processors = [
+    stylelint(),
+    reporter(
+      {
+        clearMessages: true,
+        throwError: true,
+      },
+    ),
+  ];
+
+  return gulp.src(config.files.src.sass)
+    .pipe(postcss(processors, {syntax: syntaxScss}));
+};
+
+export const testSass = () =>
```

Comment:
Does this need to be exported?

Reply:
Probably not, as it's run regularly anyway.